client/src/components/CarsFilter/index.tsx

```diff
@@ -0,0 +1,302 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import {
+  Autocomplete, TextField, Switch, Slider,
+} from '@mui/material';
+import {
+  useState, SetStateAction, useEffect,
+} from 'react';
+import './style.css';
+
+function CarsFilter() {
+  const mileRang = [
```

Comment:
no need to have these data at the component, as on every render these object will create from the strat,
SO move it to separate file and use it here,
The same for all data you used and `getYears` function